### PR TITLE
Update authorized addresses of adapters

### PR DESCRIPTION
**Pull request by Buckram123**
New method for `abstract_client::Application` "authorize_on_adapters" for authorizing application on adapters
### Checklist

- [ ] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-client/src/account.rs [hunk 1]
```diff
@@ -403,7 +403,7 @@ impl<Chain: CwEnv> Account<Chain> {
 
     /// Install an application on the account.
     /// if `install_on_sub_account` is `true`, the application will be installed on new a sub-account. (default)
-    pub fn install_app<M: InstallConfig + From<Contract<Chain>>>(
+    pub fn install_app<M: ContractInstance<Chain> + InstallConfig + From<Contract<Chain>>>(
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply framework/packages/abstract-client/src/account.rs:406]: > Why did you need to re-add this trait bound?

For `.address` in application methods

File: framework/packages/abstract-client/src/application.rs [hunk 2]
```diff
@@ -32,7 +32,7 @@ impl<Chain: CwEnv, M> DerefMut for Application<Chain, M> {
     }
 }
 
-impl<Chain: CwEnv, M: RegisteredModule> Application<Chain, M> {
+impl<Chain: CwEnv, M: RegisteredModule + ContractInstance<Chain>> Application<Chain, M> {
```
Comment (CyberHoward) [inline framework/packages/abstract-client/src/application.rs:35]: If you only need that bound in the `authorize_on_adapters` you can create a new impl with that bound added to it and leave this impl without it.

Alternatively we could add a supertrait that overwraps these traits so the signatures are smaller.

File: framework/packages/abstract-client/src/account.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Buckram123) [reply framework/packages/abstract-client/src/account.rs]: Fixed

File: framework/packages/abstract-client/src/application.rs (hunk 2 above)
Comment (Buckram123) [reply framework/packages/abstract-client/src/application.rs]: Fixed

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**